Tests/Functional/OldModel/AbstractTimeSpanTest.php

```diff
@@ -111,4 +146,260 @@ public function getTimeForBeginTimeAndEndTimeOnDifferentDaysReturnsBothTimesWith
             $this->subject->getTime()
         );
     }
+
+    /**
+     * @test
+     */
+    public function getTimeForNoTimeNotCallsHook()
+    {
+        $hook = $this->createMock(TimeSpan::class);
+        $hook->expects(self::never())->method('modifyDateSpan');
+        $hook->expects(self::never())->method('modifyTimeSpan');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TimeSpan::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
```

Comment:
I think we also need the `getInstance` calls in `tearDown` to get rid of the instances.

Reply:
IIRC, no. Nimut and the Typo3 testing framework do a reset of the instance buffer (and a lot more to start with a fresh Typo3 instance) before each test. Its all in `setUp()` and the main reason why it takes endless amounts of time to run the tests (and why there is a DB copy left after the tests end).

Reply:
Just tested: This is not required.